lib/frontend/src/switch.rs

```diff
@@ -0,0 +1,193 @@
+use std::collections::HashMap;
+use cranelift_codegen::ir::*;
+use cranelift_codegen::ir::condcodes::IntCC;
+use frontend::FunctionBuilder;
+
+type EntryIndex = i64;
+
+#[derive(Debug)]
+pub struct Switch {
+    cases: HashMap<EntryIndex, Ebb>
+}
+
+impl Switch {
+    pub fn new() -> Self {
+        Switch {
+            cases: HashMap::new(),
+        }
+    }
+
+    pub fn set_entry(&mut self, index: EntryIndex, ebb: Ebb) {
+        assert!(self.cases.insert(index, ebb).is_none(), "Tried to set the same entry {} twice", index);
+    }
+
+    fn build_cases_tree(self) -> Vec<(EntryIndex, Vec<Ebb>)> {
+        println!("before: {:#?}", self.cases);
```

Comment:
We are using the `log` crate now, so you could make these use `debug!` for logging this kind of information.

Reply:
Sure, I forgot that i added those.